mirgecom/logging_quantities.py

```diff
@@ -310,6 +310,25 @@ def __call__(self) -> float:
         return get_inviscid_cfl(self.discr, eos=self.eos, dt=self.dt, cv=self.state)
 
 
+class LogUserQuantity(LogQuantity):
+    """Logging support for arbitrary user quantities."""
+
+    def __init__(self, name="user_quantity", value=None, user_function=None) -> None:
```

Comment:
While the user_function is not needed for this particular quantity, it would be required for doing anything more complicated than just setting a simple value.  This interface enables the user to specify a `F(self._quantity)` instead of a simple scalar value, without the need to extend logging with another user-defined quantity class.

Reply:
I think passing in a function here might be a bit dangerous - how do you pass the function arguments, and how do you make sure they aren't stale? I'd remove this, and maybe think about creating a new class whenever we have a more clear use case.

Reply:
```suggestion
    def __init__(self, name, unit, description="") -> None:
```

Reply:
I think the burden of implementing the function with all its required data is on the user.  This will be needed for anything more complicated than a simple value set.  Consider this example, where the user wanted instead to perform the reduction on the *inside* of the logging class: 

```python
def get_domain_max(discr, field):
  from grudge.op import nodal_max
  return nodal_max(discr, "vol", field)

logmgr.add_quantity(UserLogQuantity(name="cfl", value=current_cfl,
                    description="inviscid CFL",
                    user_function=partial(get_domain_max, discr))

def pre_step_loop_body(step, t, dt, state):
  if do_viz:
   current_cfl = get_inviscid_cfl(blah, blah, blah)
   logmgr.set_quantity_value("cfl", current_cfl)
```

To me, this code is better than what we are doing currently (even in this PR) and is superior to the one where the discretization object (or the state, and/or the EOS) are implicitly used to calculate quantities under-the-covers.